game/src/ecs/components/ai/fight/FightAI.java

```diff
@@ -0,0 +1,8 @@
+package ecs.components.ai.fight;
+
+import ecs.entities.Entity;
+
+public interface FightAI {
```

Comment:
Sollten Interfaces nicht über den Namen erkennbar sein, also hier sowas wie `IFightAI`?

Reply:
Pass auf was du sagst, mittlerweile ist das großes Diskussionsthema

Reply:
gefixed, haben wir sonst auch überall gemacht und dann halt ich mich daran

Reply:
> Pass auf was du sagst, mittlerweile ist das großes Diskussionsthema

Gibt mal mehr Kontext - hab ich was verpasst? Also Diskussion unter euch oder in "der Welt" (Quellen)?

Ich bin eigentlich auch kein großer Fan von der Kennzeichnung Interfaces vs. "normale" Klassen/Typen. Aber grad hier sind Interfaces ja definitiv was besonderes, da würde ich sie gern schnell/einfach/kanonisch erkennen (und nicht erst in der IDE beim Mouse-Over). Aber vielleicht gibt es gute Argumente, das anders zu machen? Bin gespannt :)

Reply:
> Gibt mal mehr Kontext - hab ich was verpasst? Also Diskussion unter euch oder in "der Welt" (Quellen)?

In der Welt. Hab das schon in mehreren Foren etc. gesehen und scheint sich auch in der jüngeren Literatur wiederzufinden. 

Mal als Beispiel (habs mal überflogen und so die typischen Argumente beider Seiten wiedergefunden): https://softwareengineering.stackexchange.com/questions/117348/should-interface-names-begin-with-an-i-prefix


Ich neige mittlerweile auch zum weglassen des Präfix, da es eigentlich keinen Mehrwert liefert und nur "rauschen" ist. Allerdings hab ich es mit Präfix gelernt und mach es unterbewusst noch sehr oft. Und Consistency ist halt irgendwie wichtiger.

Reply:
Hmmm, die verlinkte Diskussion ist aber auch nicht wirklich "recent", der als Lösung markierte Kommentar ist von 2011 :)

Aus User-Sicht ist das zusätzliche "I" tatsächlich nur Rauschen. Für uns ist es aber schon eine Information, dass wir gewissermaßen einen Obertyp nutzen/definieren, sprich dass es in der Hierarchie etwas heraussticht. (bis hier hin meine Meinung, die sich aber mit dem Kommentar deckt) Im Kommentar wurde noch auf mögliche Vorteile bei der Nutzung mit/in Reflection hingewiesen, was bei uns ja leider der Fall ich ist.

Reply:
Fachliteratur sollte in den Sachmitteln drin sein. 

Sammle gern Vorschläge (mach ein Ticket auf) und wir schauen dann. 

So Klassiker wie Clean Code sollte es imho in der Bibliothek geben (oder in Deiner Privatbibliothek), und zu unserem eigentlichen Projektthema gibt es vermutlich gar keine Bücher, sondern man muss sich passende Paper zusammen suchen.